common/app/views/support/cleaner/AmpEmbedCleaner.scala

```diff
@@ -60,6 +60,33 @@ case class AmpEmbedCleaner(article: Article) extends HtmlCleaner {
 
   }
 
+  object AmpSoundcloud {
+    def createElement(document: Document, trackId: String): Element = {
+      val soundcloud = document.createElement("amp-soundcloud")
+      soundcloud.attr("data-trackid", trackId)
+      soundcloud.attr("data-visual", "true")
+      soundcloud.attr("height", "300") // height is necessary if data-visual == true
+    }
+
+    def getTrackIdFromUrl(soundcloudUrl: String): String = {
+      val pattern = "soundcloud.com%2Ftracks%2F(\\d+)".r
+      pattern.findAllIn(soundcloudUrl).matchData.map { matches => matches.group(1) }.mkString
+    }
+
+    def clean(document: Document) = {
+      document.getElementsByClass("element-audio").foreach { audioElementWithIframe: Element =>
+        audioElementWithIframe.getElementsByTag("iframe").foreach { iframeElement: Element =>
+          val soundcloudUrl = iframeElement.attr("src")
+          Option(getTrackIdFromUrl(soundcloudUrl)) filterNot { _.isEmpty } foreach { trackId =>
```

Comment:
Damn, that's how I originally implemented it. I should have trusted my instincts! 😄 

Thanks for your help, dude